Userland/Libraries/LibWeb/HTML/HTMLMediaElement.cpp

```diff
@@ -62,4 +94,544 @@ void HTMLMediaElement::pause() const
+            //           ability to cache data.
+
+            // 5. Otherwise, incrementally read response's body given updateMedia, processEndOfMedia, an empty algorithm, and global.
+            VERIFY(response->body().has_value());
+            auto empty_algorithm = [](auto&) {};
+
+            // FIXME: We are "fully" reading the response here, rather than "incrementally". Memory concerns aside, this should be okay for now as we are
+            //        always setting byteRange to "entire resource". However, we should switch to incremental reads when that is implemented, and then
+            //        implement the processEndOfMedia step.
+            response->body()->fully_read(realm, move(update_media), move(empty_algorithm), JS::NonnullGCPtr { global }).release_value_but_fixme_should_propagate_errors();
+        };
+
+        m_fetch_controller = TRY(Fetch::Fetching::fetch(realm, request, Fetch::Infrastructure::FetchAlgorithms::create(vm, move(fetch_algorithms_input))));
+        break;
+    }
+
+    // -> Otherwise (mode is local)
+    case FetchMode::Local:
+        // FIXME:
+        // The resource described by the current media resource, if any, contains the media data. It is CORS-same-origin.
+        //
+        // If the current media resource is a raw data stream (e.g. from a File object), then to determine the format of the media resource, the user agent
+        // must use the rules for sniffing audio and video specifically. Otherwise, if the data stream is pre-decoded, then the format is the format given
+        // by the relevant specification.
+        //
+        // Whenever new data for the current media resource becomes available, queue a media element task given the media element to run the first appropriate
+        // steps from the media data processing steps list below.
+        //
+        // When the current media resource is permanently exhausted (e.g. all the bytes of a Blob have been processed), if there were no decoding errors,
+        // then the user agent must move on to the final step below. This might never happen, e.g. if the current media resource is a MediaStream.
+        break;
+    }
+
+    return {};
+}
+
+// https://html.spec.whatwg.org/multipage/media.html#verify-a-media-response
+bool HTMLMediaElement::verify_response(JS::NonnullGCPtr<Fetch::Infrastructure::Response> response, ByteRange const& byte_range)
+{
+    // 1. If response is a network error, then return false.
+    if (response->is_network_error())
+        return false;
+
+    // 2. If byteRange is "entire resource", then return true.
+    if (byte_range.has<EntireResource>())
+        return true;
+
+    // 3. Let internalResponse be response's unsafe response.
+    // 4. If internalResponse's status is 200, then return true.
+    // 5. If internalResponse's status is not 206, then return false.
+    // 6. If the result of extracting content-range values from internalResponse is failure, then return false.
+    TODO();
```

Comment:
Why can't we implement this yet?

Reply:
I only implemented the `"entire resource"` case above, so anything implemented here is unreachable. Hence the `TODO` call instead of a FIXME, if we hit this at all, it's because someone half-implemented other cases but forgot to then implement this.